### PR TITLE
Let FiniteElement::get_*data return a std::unique_ptr

### PR DESCRIPTION
**Pull request by masterleinad**
Related to #5880.

### CODE REVIEW

**Comment by bangerth** [comment]
The build seems to have failed. Can you take a look?

/run-tests

**Comment by bangerth** [comment]
Actually, if you could review #6114, then I can post the corresponding patch to yours, just for the mappings. This required a lot more work than I thought due to the fact that we used covariant return types everywhere.

But I have that part figured out, compiling, and testing correctly. If you're willing to wait for that patch, you can see how I solved things.

File: include/deal.II/base/thread_management.h [hunk 1]
```diff
@@ -735,6 +735,37 @@ namespace Threads
...
+     * @internal
+     *
+     * Given an arbitrary type RT, store an element of it and grant access to
+     * it through functions get() and set(). This is the specialization for
+     * std::unique_ptr types: since we can't copy them only allow to call get()
+     * one time after set() has been called.
+     */
+    template <typename RT> struct return_value<std::unique_ptr<RT> >
+    {
+    private:
```
Comment (bangerth) [inline include/deal.II/base/thread_management.h]: Nah, let's not go there. #6114 has the more general solution.

**Comment by bangerth** [comment]
Take a look at #6115, which solves some of the problems you're running into with the compiler error. If you don't mind, let's merge #6114 and #6115 first, and then this should just fall into place!

**Comment by masterleinad** [comment]
I rebased to use #6114.

**Pull request merged**